src/ec2/model.ts

```diff
@@ -41,14 +41,17 @@ export class Ec2ConnectionManager {
         return new DefaultIamClient(this.regionCode)
     }
 
-    protected async getAttachedPolicies(instanceId: string): Promise<IAM.attachedPoliciesListType> {
+    protected async getAttachedPolicies(instanceId: string): Promise<IAM.AttachedPolicy[]> {
         const IamRole = await this.ec2Client.getAttachedIamRole(instanceId)
         if (!IamRole) {
             return []
         }
-        const iamResponse = await this.iamClient.listAttachedRolePolicies(IamRole!.Arn!)
-
-        return iamResponse.AttachedPolicies ?? []
+        try {
+            const attachedRoles = await this.iamClient.listAttachedRolePolicies(IamRole!.Arn!)
```

Comment:
In the docstring for `getAttachedPolicies` we could document that "error Foo is raised if IAM could not find the role". Then the caller can handle that error. This would allow us to show a better error message to the user.

Alternatively we could throw a ToolkitError. That is probably more idiomatic and also will be automatically handled even if the caller doesn't handle it. It also serves as a kind of documentation.

Reply:
changed it to throw a ToolkitError so that it bubbles the error up a level. Now, it logs what happened, but still assumes necessary policies don't exist.